src/Riok.Mapperly/MapperGenerator.cs

```diff
@@ -22,39 +22,67 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
         var mapperClassDeclarations = SyntaxProvider.GetClassDeclarations(context);
 
         var compilationAndMappers = context.CompilationProvider.Combine(mapperClassDeclarations.Collect());
-        context.RegisterImplementationSourceOutput(compilationAndMappers, static (spc, source) => Execute(source.Left, source.Right, spc));
+        IncrementalValueProvider<(
+            ImmutableArray<(MemberDeclarationSyntax, string)> Mappers,
+            ImmutableArray<Diagnostic> Diagnostics
+        )> mappersWithErrors = compilationAndMappers.Select((x, token) => BuildDescriptors(x.Left, x.Right, token));
```

Comment:
I think it would make sense to create a class for the Mappers / Diagnostic combination.